### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by FadiAlamassi**
#31

### CODE REVIEW

File: client/src/components/pages/Home/ReportBuilding/Form/index.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        preferredUse: 'N/A',
+        emptyPeriod: '1 year',
+        extraInfo: 'more info about this building',
+        approved: true,
+        receiveNotifications: false,
+        reporterName: 'Som',
+        reporterEmail: 'ahmed@gmail.com',
+        reporterAddress: 'gaza',
+      };
+      await axios.post('/api/v1/report-building', dummy);
```
Comment (MaiUbeid) [inline client/src/components/pages/Home/ReportBuilding/Form/index.js]: I added dummy object to check Axios request but it gives 400 bad request 
Can you check it ? 
@ahmedisam99

File: client/src/components/pages/Home/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  handleAddressChange = (markerCoordinates, address) => {
+    if (markerCoordinates && address) {
+      const { lat: latitude, lng: longitude } = markerCoordinates;
+      this.setState({ latitude, longitude, address });
+    }
+  };
+
+  handleCityChange = ({ key }) => {
+    const city = +key === 1 ? 'Morecambe' : 'Hastings';
+    this.setState({ city });
```
Comment (ahmedisam99) [inline client/src/components/pages/Home/index.js]: just to always make sure, check for the `+key` value here:
```js
if(+key) {
    const city = +key === 1 ? 'Morecambe' : 'Hastings';
    this.setState({ city });
}
```

File: client/src/components/pages/Home/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <About />
+        <ReportBuilding
+          longitude={longitude}
+          latitude={latitude}
+          redirectToView={this.redirectToView}
+          city={city}
+          address={address}
+          markerCoordinates={{ lat: latitude, lng: longitude }}
+          onCityChange={this.handleCityChange}
+          handleAddressChange={this.handleAddressChange}
```
Comment (ahmedisam99) [inline client/src/components/pages/Home/index.js]: Just a suggestion, as long as you pass all the values from the state with the exact same key names, why don't you just spread the state here...
```jsx
        <ReportBuilding
          {...this.state}
          redirectToView={this.redirectToView}
          markerCoordinates={{ lat: latitude, lng: longitude }}
          onCityChange={this.handleCityChange}
          handleAddressChange={this.handleAddressChange} 
        />
```

File: client/src/components/pages/Home/ReportBuilding/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      .run((error, result) => {
+        if (error) {
+          handleAddressChange(undefined, undefined);
+          openNotificationWithIcon(
+            'error',
+            'Something went wrong !! Check your connection and try again',
+          );
+          return;
+        }
+        handleAddressChange(latlng, result.address.Match_addr);
```
Comment (ahmedisam99) [inline client/src/components/pages/Home/ReportBuilding/index.js]: calling `handleAddressChange` with `undefined` for both values will cause the function to just not do anything, so why call it in the first place.

```js
        if (error)
          openNotificationWithIcon('error', 'Something went wrong!! Check your connection and try again');
        else handleAddressChange(latlng, result.address.Match_addr);
```

File: client/src/components/pages/Home/ReportBuilding/Map/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          url="https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}"
+          id="mapbox.streets"
+          accessToken="pk.eyJ1IjoiZmFkeW1hemVuIiwiYSI6ImNrMXRxY3JjZDBpMGYzbXF5YmV0c2g5a24ifQ.C6ZZQF61IMwDVQmu7Xxpzg"
+        />
+        <Polygon
+          positions={cityBoundary}
+          fillOpacity={0}
+          onclick={onCityClick}
+        />
+        {markerCoordinates !== undefined &&
```
Comment (ahmedisam99) [inline client/src/components/pages/Home/ReportBuilding/Map/index.js]: if I tried and manipulated the value of either latitude or longitude and made it a string, for example, the whole app will crash 
```suggestion
        {markerCoordinates && markerCoordinates.lat && markerCoordinates.lng &&
```

File: client/src/components/pages/Home/index.js (hunk 3 above)
Comment (Amoodaa) [reply client/src/components/pages/Home/index.js]: i think spreading props is forbidden? https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md

Comment (MaiUbeid) [reply client/src/components/pages/Home/index.js]: @ahmedisam99 
 when I passed it as spread it didn't work well, and it caused confusion for me !!
Did u try it in the code?

File: client/src/components/pages/Home/ReportBuilding/Form/index.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+      longitude,
+      latitude,
+      emptyPeriod,
+      extraInfo,
+      preferredUse,
+      ...stepThreeValues,
+    };
+
+    formData.append('data', JSON.stringify(building));
+    formData.append('thumbnail', thumbnail[0].originFileObj);
```
Comment (ahmedisam99) [inline client/src/components/pages/Home/ReportBuilding/Form/index.js]:
```suggestion
    if(thumbnail && thumbnail[0])
        formData.append('thumbnail', thumbnail[0].originFileObj);
```

Comment (MaiUbeid) [reply client/src/components/pages/Home/ReportBuilding/Form/index.js]: I did it but forget to push it 
I will push it ^_^
@ahmedisam99

File: client/src/components/pages/Home/index.js (hunk 3 above)
Comment (ahmedisam99) [reply client/src/components/pages/Home/index.js]: @Amoodaa 
we're above the law :joy:

**Pull request merged**